internal/reader/rmn_home.go

```diff
@@ -0,0 +1,311 @@
+package reader
+
+import (
+	"context"
+	"crypto/ed25519"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	mapset "github.com/deckarep/golang-set/v2"
+
+	"github.com/smartcontractkit/chainlink-ccip/pkg/consts"
+
+	"github.com/smartcontractkit/chainlink-ccip/internal/reader/contractreader"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+
+	rmntypes "github.com/smartcontractkit/chainlink-ccip/commit/merkleroot/rmn/types"
+)
+
+const (
+	rmnMaxSizeCommittee = 256 // bitmap is 256 bits making the max committee size 256
+)
+
+type RMNHome interface {
+	// GetRMNNodesInfo gets the RMNHomeNodeInfo for the given configDigest
+	GetRMNNodesInfo(configDigest cciptypes.Bytes32) ([]rmntypes.RMNHomeNodeInfo, error)
+	// IsRMNHomeConfigDigestSet checks if the configDigest is set in the RMNHome contract
+	IsRMNHomeConfigDigestSet(configDigest cciptypes.Bytes32) (bool, error)
+	// GetMinObservers gets the minimum number of observers required for each chain in the given configDigest
+	GetMinObservers(configDigest cciptypes.Bytes32) (map[cciptypes.ChainSelector]uint64, error)
+	// GetOffChainConfig gets the offchain config for the given configDigest
+	GetOffChainConfig(configDigest cciptypes.Bytes32) (cciptypes.Bytes, error)
+	services.Service
+}
+
+type rmnHomeState struct {
+	primaryConfigDigest   cciptypes.Bytes32
+	secondaryConfigDigest cciptypes.Bytes32
+	rmnHomeConfig         map[cciptypes.Bytes32]rmntypes.RMNHomeConfig
+}
+
+type RmnHomePoller struct {
+	wg                   sync.WaitGroup
+	stopCh               services.StopChan
+	sync                 services.StateMachine
+	contractReader       contractreader.ContractReaderFacade
+	rmnHomeBoundContract types.BoundContract
+	lggr                 logger.Logger
+	mutex                *sync.RWMutex
+	rmnHomeState         rmnHomeState
+	failedPolls          uint
+	pollingDuration      time.Duration // How frequently the poller fetches the chain configs
+}
+
+func NewRMNHomePoller(
+	contractReader contractreader.ContractReaderFacade,
+	rmnHomeBoundContract types.BoundContract,
+	lggr logger.Logger,
+	pollingInterval time.Duration,
+) RMNHome {
+	return &RmnHomePoller{
+		stopCh:               make(chan struct{}),
+		contractReader:       contractReader,
+		rmnHomeBoundContract: rmnHomeBoundContract,
+		rmnHomeState:         rmnHomeState{},
+		mutex:                &sync.RWMutex{},
+		failedPolls:          0,
+		lggr:                 lggr,
+		pollingDuration:      pollingInterval,
+	}
+}
+
+func (r *RmnHomePoller) Start(ctx context.Context) error {
+	r.lggr.Infow("Start Polling RMNHome")
+	return r.sync.StartOnce(r.Name(), func() error {
```

Comment:
I am not familiar with `services.StateMachine`, what happens if we call `Start` twice? returns an error?

Reply:
Exactly we have a check one the state to verify that has never been started